src/main.js

```diff
@@ -1053,9 +1054,9 @@ const DiscoveryFeedAppStoreLinkCard = new Lang.Class({
             })
         });
         this.add(card);
-        card.connect('clicked', Lang.bind(this, function() {
+        card.connect('button-release-event', Lang.bind(this, function(widget, event) {
```

Comment:
It's not needed to change this -- you should use `clicked` instead, as it handles things like double click thresholds/etc internally.
You should be able to use `gtk_get_current_event_time()` in the `clicked` callback to achieve the same effect.

Reply:
Hmm, okay, handy. It isn't relevant here but I wonder how Gtk deals with multithreading in this case. I guess it doesn't.

Reply:
Yeah, GTK code is not supposed to be called into from multiple threads.